plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt

```diff
@@ -265,7 +275,23 @@ private class AmazonQServerInstance(private val project: Project, private val cs
         launcherHandler.addProcessListener(inputWrapper)
         launcherHandler.startNotify()
 
-        launcher = LSPLauncher.Builder<AmazonQLanguageServer>()
+        class AmazonQServerBuilder : Builder<AmazonQLanguageServer>() {
+            private val customMessageTracer = MessageTracer()
+
+            override fun wrapMessageConsumer(consumer: MessageConsumer?): MessageConsumer =
+                super.wrapMessageConsumer { message ->
+                    customMessageTracer.trace("INCOMING", message)
+                    if (message is ResponseMessage && message.result is AwsExtendedInitializeResult) {
+                        val result = message.result as AwsExtendedInitializeResult
+                        AwsServerCapabilitiesProvider.getInstance(project).setAwsServerCapabilities(result.getAwsServerCapabilities())
```

Comment:
we already expose capabilities on the service
```
AmazonQLspServer.getInstance(project).capabilities.serverCapabilities
```
?

Reply:
Want to keep this in place to be able to access other agents potentially from this map